lib/fog/libvirt/requests/compute/list_domains.rb

```diff
@@ -48,6 +48,23 @@ def boot_order xml
           xml_elements(xml, "domain/os/boot", "dev")
         end
 
+        # Foreman expects the firmware to be 'uefi_sb' if SB is enabled
+        def firmware(xml)
+          firmware_type = xml_elements(xml, "domain/os", "firmware").first || 'bios'
+          return 'uefi_sb' if firmware_type == 'efi' && secure_boot_enabled?(xml)
+
+          firmware_type
+        end
```

Comment:
I think `firmware` should exactly output what was also input as an attribute. So I'd just expect `xml_elements(xml, "domain/os", "firmware")`. Then you also read `firmware_features` back and set that attribute. That keeps it consistent that when you create a VM with certain attributes, you can read them back exactly as they were.

In other words, there's a `to_xml` and this can be seen as a `from_xml`.